File: Components/DevExpressVCL/Demos/ExpressSpreadSheet/CBuilder/SimpleReport/SimpleReportDemoMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "SimpleReportDemoMain.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "BaseForm"
#pragma link "ReportDesignerBaseForm"
#pragma link "ReportPreviewUnit"
#pragma link "cxClasses"
#pragma link "cxLookAndFeels"
#pragma link "cxCheckBox"
#pragma link "cxContainer"
#pragma link "cxControls"
#pragma link "cxEdit"
#pragma link "cxGraphics"
#pragma link "cxGroupBox"
#pragma link "cxLabel"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxSplitter"
#pragma link "cxTextEdit"
#pragma link "dxSpreadSheetCore"
#pragma link "dxSpreadSheetReportDesigner"
#pragma link "cxFilterControl"
#pragma link "ReportDesignerBaseForm"
#pragma link "dxmdaset"
#pragma resource "*.dfm"
TfrmSimpleReport *frmSimpleReport;

__fastcall TfrmSimpleReport::TfrmSimpleReport(TComponent* Owner)
	: TfrmReportDesignerBase(Owner)
{
}

//---------------------------------------------------------------------------

void TfrmSimpleReport::Initialize()
{
  LoadDataset(mdsOrders, "..\\..\\Data\\OrderDetails.mds");
  ReportDesigner->LoadFromFile("..\\..\\Data\\SimpleReportTemplate.xlsx");
  TfrmReportDesignerBase::Initialize();
}

File: Components/DevExpressVCL/Demos/ExpressSpreadSheet/Delphi/SimpleReport/SimpleReportDemoMain.pas
unit SimpleReportDemoMain;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms, Dialogs, ReportDesignerBaseForm, cxGraphics, cxControls, cxLookAndFeels,
  cxLookAndFeelPainters, cxContainer, cxEdit, cxClasses, Menus, StdCtrls, cxFilterControl, cxCheckBox,  cxGroupBox, ExtCtrls, cxSplitter, dxSpreadSheetCore,
  dxSpreadSheetReportDesigner, dxSpreadSheet, DB, dxmdaset, ComCtrls;


type
  TfrmSimpleReport = class(TfrmReportDesignerBase)
    dsOrders: TDataSource;
    mdsOrders: TdxMemData;
  private
    { Private declarations }
  public
    procedure BeforePreview(ASpreadSheet: TdxSpreadSheet); override;
    procedure Initialize; override;
  end;

var
  frmSimpleReport: TfrmSimpleReport;

implementation

{$R *.dfm}

{ TfrmSimpleReport }

procedure TfrmSimpleReport.BeforePreview(ASpreadSheet: TdxSpreadSheet);
begin
  //
end;

procedure TfrmSimpleReport.Initialize;
begin
  LoadDataset(mdsOrders, '..\..\Data\OrderDetails.mds');
  ReportDesigner.LoadFromFile('..\..\Data\SimpleReportTemplate.xlsx');
  inherited Initialize;
end;

end.
